qiskit_nature/drivers/molecule.py

```diff
@@ -71,6 +76,57 @@ def __init__(
 
         self._perturbations = None  # type: Optional[List[float]]
 
+    def to_hdf5(self, parent: h5py.Group) -> None:
+        """Stores this instance in a HDF5 group inside of the provided parent group.
+
+        Args:
+            parent: the parent HDF5 group.
+        """
+        group = parent.require_group(self.__class__.__name__)
+        group.attrs["__class__"] = self.__class__.__name__
+        group.attrs["__module__"] = self.__class__.__module__
+        group.attrs["__version__"] = self.VERSION
+
+        geometry_group = group.create_group("geometry", track_order=True)
+        for idx, geom in enumerate(self._geometry):
+            symbol, coords = geom
+            geometry_group.create_dataset(str(idx), data=coords)
+            geometry_group[str(idx)].attrs["symbol"] = symbol
+
+        group.attrs["multiplicity"] = self._multiplicity
+        group.attrs["charge"] = self._charge
```

Comment:
Yes, indeed, this will need to be aligned with #481, once that gets merged. However, since that PR is stalling I would like to get this one merged before.

> I will also note there is a property for units that could already be used

I do not fully understand what you mean here.

Reply:
I meant you could already do something like
```
        group.attrs["units"] = self.units
```
and since thats using the property possibly switch to using that for the others (rather than directly using the private field variable that is backing it) eg
```
        group.attrs["multiplicity"] = self.multiplicity
        group.attrs["charge"] = self.charge
 ```
 Maybe its just me, but normally I will use the public properties directly and leave the only use of the private var to the property getter/setter logic in the class.